src/modal.test.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import TestUtils from 'react-addons-test-utils';
+import Modal, { MODAL_CLOSE_BUTTON } from './Modal';
+
+describe('Modal', () => {
+
+	let modal, modalEl, spyable;
+	const content = 'Test model content';
+
+	beforeEach(() => {
+		spyable = {
+			onDismiss: (e) => {}
+		};
+
+		spyOn(spyable, 'onDismiss');
+
+		modal = TestUtils.renderIntoDocument(
+			<Modal onDismiss={spyable.onDismiss}>{content}</Modal>
+		);
+		modalEl = ReactDOM.findDOMNode(modal);
+	});
+
+	afterEach(() => {
+		modalEl = null;
+	});
+
+	it('exists', () => {
+		expect(modalEl).not.toBeNull();
+	});
+
+	it('has SQ2 modal styles', () => {
+		expect(modalEl.classList).toContain('modal');
+	});
+
+	it('displays modal content', () => {
+		expect(modalEl.innerHTML).toContain(content);
+	});
+
+	it('creates an Icon component for dismissal', () => {
+		const icon = modalEl.getElementsByClassName('svg-icon')[0];
```

Comment:
be more specific - this will match any icon that is ever part of the modal component, but you want to be able to select _only_ the icon that is used for dismissal.